include/swift/Sema/ConstraintSystem.h

```diff
@@ -3484,6 +3484,15 @@ class ConstraintSystem {
     return nullptr;
   }
 
+  Expr *getSemanticsProvidingParentExpr(Expr *expr) {
+    while (auto *parent = getParentExpr(expr)) {
+      if (!(isa<IdentityExpr>(parent) || isa<TryExpr>(parent)))
```

Comment:
Worth doing `parent->getSemanticsProvidingExpr() == parent` to avoid the duplication?